### PR TITLE
Generic text rendering

### PR DESCRIPTION
**Pull request by AJenbo**
This should gradually replace all the direct rendering of game texts
throughout the code. The interface is made to closely mirror that of the
art fonts as that is what will eventually be used for rendering Unicode
fonts both in the menus and in-game.

Performance before 1082fps and after 1142fps
(my FPS isn't super stable so your mileage may vary)

### CODE REVIEW

File: Source/engine/render/automap_render.cpp [hunk 1]
```diff
@@ -1,3 +1,8 @@
+/**
+ * @file automap_render.cpp
+ *
+ * Sprite animation handeling.
```
Comment (glebm) [inline Source/engine/render/automap_render.cpp:4]: This comment looks wrong. Moreover, why even have file comments for `cpp` files when there is a corresponding header?

Comment (AJenbo) [reply Source/engine/render/automap_render.cpp:4]: Doxygen doesn't pickup files unless they have a header :/

Comment (AJenbo) [reply Source/engine/render/automap_render.cpp]: I suppose we could see this as a feature and only pickup public header functions variables etc?

Comment (glebm) [reply Source/engine/render/automap_render.cpp]: Yeah, I think it's a feature -- the stuff in cpp files is implementation details

**Review by glebm — APPROVED** [review]

**Pull request merged**